File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        if(nums == null || nums.length == 0){
            return new int[]{-1,-1};
        }

        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i = 0; i< nums.length; i++){
            int diff = target-nums[i];
            if(map.containsKey(diff)){
                return new int[]{map.get(diff),i};
            }
            map.put(nums[i],i);
        }

        return new int[]{-1,-1};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}
        for i in range (len(nums)):
            diff = target - nums[i]
            if diff in map:
                return [i,map[diff]]
            else:
                map[nums[i]] = i
        